monitor/xfeemngr/xfeemngr.go

```diff
@@ -0,0 +1,110 @@
+package xfeemngr
+
+import (
+	"context"
+	"crypto/ecdsa"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/tokens"
+	"github.com/omni-network/omni/lib/tokens/coingecko"
+	"github.com/omni-network/omni/lib/xchain"
+	"github.com/omni-network/omni/monitor/xfeemngr/gasprice"
+	"github.com/omni-network/omni/monitor/xfeemngr/ticker"
+	"github.com/omni-network/omni/monitor/xfeemngr/tokenprice"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type Manager struct {
+	gprice  *gasprice.Buffer
+	tprice  *tokenprice.Buffer
+	oracles map[uint64]feeOracle
+	ticker  ticker.Ticker
+}
+
+func Start(ctx context.Context, network netconf.Network, endpoints xchain.RPCEndpoints, privKeyPath string) error {
+	privKey, err := crypto.LoadECDSA(privKeyPath)
+	if err != nil {
+		return errors.Wrap(err, "load private key")
+	}
+
+	gasPricers, err := makeGasPricers(network, endpoints)
+	if err != nil {
+		return err
+	}
+
+	gprice := gasprice.NewBuffer(gasPricers)
+	tprice := tokenprice.NewBuffer(coingecko.New(), tokens.OMNI, tokens.ETH)
+
+	oracles, err := makeOracles(ctx, network, endpoints, privKey, gprice, tprice)
+	if err != nil {
+		return err
+	}
+
+	m := Manager{
+		gprice:  gprice,
+		tprice:  tprice,
+		oracles: oracles,
+		ticker:  ticker.New(),
+	}
+
+	go m.start(ctx)
```

Comment:
When will this goroutine stop? Does it stop if the context is cancelled?

Reply:
Yes